docs/content/docs/ops/batch/recovery_from_job_master_failure.md

```diff
@@ -0,0 +1,99 @@
+---
+title: "Recovery from job master failures"
+weight: 4
+type: docs
+aliases:
+
+- /docs/ops/batch/recovery_from_job_master_failure.html
+- /docs/ops/batch/recovery_from_job_master_failure
+
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Batch jobs Recovery from job master failures
+
+## Background
+
+Currently, if the JobMaster fails and is terminated, one of the following two situations will occur:
+
+- If high availability (HA) is disabled, the job will fail.
+- If HA is enabled, a JobMaster failover will happen and the job will be restarted. For streaming jobs, it can resume 
+  from the last successful checkpoint. However, for batch jobs, which do not have checkpoints, it will have to start
+  over from the beginning, losing all previously made progress. This represents a significant regression for
+  long-running batch jobs.
+
+To address this issue, we introduced a feature for the recovery of batch jobs after a JobMaster failover in Flink 
+version 1.20. The main purpose of this feature is to enable batch jobs to recover as much progress as possible after a JobMaster
```

Comment:
To address this issue, we introduced a feature for the recovery of batch jobs after a JobMaster failover in Flink 
version 1.20. The main purpose of this feature is to enable ...

-> To address this issue, a batch job recovery mechanism is introduced to enable ...